frontend/src/components/codes/Codes.js

```diff
@@ -1,122 +1,7 @@
-
-  const userContext = useContext(UserContext);
-  const [state, setState] = useState({
-    loading: true,
-    data: [],
-    selectedUri: null,
-    deleteDialogTitle: '',
-    showDeleteDialog: false,
-  });
-  const [trigger, setTrigger] = useState(true);
-
-  useEffect(() => {
-    fetchDataTypes('code').then(res => {
-      if(res.success)
-        setState(state => ({...state, loading: false, data: res.codes}));
-    }).catch(e => {
-      setState(state => ({...state, loading: false}))
-      navigate('/dashboard');
-      enqueueSnackbar(e.json?.message || "Error occur", {variant: 'error'});
-    });
-  }, [trigger]);
-
-  const showDeleteDialog = (uri) => {
-    setState(state => ({
-      ...state, selectedUri: uri, showDeleteDialog: true,
-      deleteDialogTitle: 'Delete codes ' + uri + ' ?'
-    }));
-  };
-
-  const handleDelete = async (uri, form) => {
-
-    deleteTheme(uri).then(({success, message})=>{
-      if (success) {
-        setState(state => ({
-          ...state, showDeleteDialog: false,
-        }));
-        setTrigger(!trigger);
-        enqueueSnackbar(message || "Success", {variant: 'success'})
-      }
-    }).catch((e)=>{
-      setState(state => ({
-        ...state, showDeleteDialog: false,
-      }));
-      reportErrorToBackend(e)
-      setTrigger(!trigger);
-      enqueueSnackbar(e.json?.message || "Error occur", {variant: 'error'});
-    });
+import CodeView from "./CodeView";
 
-  };
-
-  const columns = [
-    {
-      label: 'Name',
-      body: ({_uri, name}) => {
-        return <Link colorWithHover to={`/code/${encodeURIComponent(_uri)}/view`}>
-          {name}
-        </Link>
-      },
-      sortBy: ({name}) => name
-    },
-    {
-      label: 'Description',
-      body: ({description}) => {
-        return description;
-      }
-    },
-
-    {
-      label: ' ',
-      body: ({_uri}) =>
-        <DropdownMenu urlPrefix={'code'} objectUri={encodeURIComponent(_uri)} hideDeleteOption
-                      hideEditOption={!userContext.isSuperuser} handleDelete={() => showDeleteDialog(_uri)}/>
-    }
-  ];
-
-  if (state.loading)
-    return <Loading message={`Loading codes...`}/>;
-
-  return (
-    <Container>
-      <DataTable
-        title={"Codes"}
-        data={state.data}
-        columns={columns}
-        uriField="uriField"
-        customToolbar={
-          <Chip
-            disabled={!userContext.isSuperuser}
-            onClick={() => navigate('/code/new')}
-            color="primary"
-            icon={<AddIcon/>}
-            label="Add new Codes"
-            variant="outlined"/>
-        }
-
-      />
-      <DeleteModal
-        objectUri={state.selectedUri}
-        title={state.deleteDialogTitle}
-        show={state.showDeleteDialog}
-        onHide={() => setState(state => ({...state, showDeleteDialog: false}))}
-        delete={handleDelete}
-      />
-    </Container>
-  );
+export default function Codes() {
+  return <CodeView codeUser multi/>
```

Comment:
It's not actually codeUser, there are 3 types of user mode, organization user, group user, and super user.
For now, we don't have to worry about their differences. But, the parameters should be organizationUser, groupUser and superuser.